test/integration_tests/test_overwrite_executor.py

```diff
@@ -0,0 +1,77 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+from pathlib import Path
+from test.util import create_sample_video, file_remove, shutdown_ray
+
+import pytest
+
+from eva.catalog.catalog_manager import CatalogManager
+from eva.configuration.constants import EVA_ROOT_DIR
+from eva.server.command_handler import execute_query_fetch_all
+
+
+@pytest.mark.notparallel
+class OverwriteExecutorTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # reset the catalog manager before running each test
+        CatalogManager().reset()
+        video_file_path = create_sample_video()
+        image_file_path = Path(
+            f"{EVA_ROOT_DIR}/test/data/uadetrac/small-data/MVI_20011/img00001.jpg"
+        )
+        udf_path = Path(f"{EVA_ROOT_DIR}/eva/udfs/ndarray/gaussian_blur.py")
+        create_udf_query = f"CREATE UDF IF NOT EXISTS GaussianBlur INPUT (frame NDARRAY UINT8(3, ANYDIM, ANYDIM)) OUTPUT (blurred_frame_array NDARRAY UINT8(3, ANYDIM, ANYDIM)) TYPE ndarray IMPL '{udf_path}';"
+        execute_query_fetch_all(create_udf_query)
+
+        query = f"LOAD VIDEO '{video_file_path}' INTO MyVideo;"
+        execute_query_fetch_all(query)
+
+        query = f"LOAD IMAGE '{image_file_path}' INTO MyImage;"
+        execute_query_fetch_all(query)
+
+    @classmethod
+    def tearDownClass(cls):
+        shutdown_ray()
+        file_remove("dummy.avi")
+        file_remove(
+            Path(
+                f"{EVA_ROOT_DIR}/test/data/uadetrac/small-data/MVI_20011/modified_img00001.jpg"
+            )
+        )
+        # clean up
+        execute_query_fetch_all("DROP TABLE IF EXISTS MyVideo;")
+        execute_query_fetch_all("DROP TABLE IF EXISTS MyImage;")
+        execute_query_fetch_all("DROP UDF IF EXISTS GaussianBlur;")
+
+    # integration test for overwrite
+    def test_should_overwrite_video(self):
+        overwrite_query = "OVERWRITE MyVideo BY GaussianBlur(data);"
+        execute_query_fetch_all(overwrite_query)
+
+        select_query = "SELECT * FROM MyVideo;"
+        actual_batch = execute_query_fetch_all(select_query)
+        modified_dir = actual_batch.column_as_numpy_array(actual_batch.columns[1])[0]
+        self.assertTrue("tmp/modified/dummy.avi" in modified_dir)
```

Comment:
We should check the video content is actually transformed by `GaussionBlur` instead of the file path. Same for the image test case.

Reply:
This integration check is designed to check if the OVERWRITE statement infrastructure works as a whole, which means checking if a new video is created and linked to the table. I'll add a check for UDF, but perhaps not as an integration test.

Reply:
Integration test should check the end-to-end is working, which is that we want the video to be transformed via `GaussionBlur`. On the other hand, the fact that we store in the `tmp/modified/dummy.avi` is one implementation option we choose (but not the only one). So checking the `tmp/modified/dummy.avi` can be unitest case for our implementation choices, and the test can use mock to avoid really loading videos and running the UDF.